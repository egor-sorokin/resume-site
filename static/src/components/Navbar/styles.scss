@import '../../styles/base/_vars.scss';

.navbar {
  width: 100%;

  &__item {
    position: relative;

    &:nth-of-type(n+2) {
      margin-top: 6px;
    }
    &:nth-of-type(2n) {
      left: 55%;
      text-align: left;
    }
    &:nth-of-type(2n+1) {
      left: -55%;
      text-align: right;
    }
  }
  &__text {
    color: $mercuryLight;
    font-family: $secondaryFont;
    font-size: 12px;
    letter-spacing: 3.4px;
    text-transform: uppercase;
    font-weight: 500;
  }
}
