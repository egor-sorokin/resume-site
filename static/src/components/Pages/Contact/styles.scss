@import '../../../styles/base/_vars.scss';

.contact-inner {
  display: flex;
  flex-direction: column;

  margin: 0 auto;

  width: 30%;
  height: 100%;
  align-items: center;

  &__itme {
    display: flex;
    position: relative;
  }
}

.contact {
  .logo {
    margin-top: 90px;
    margin-bottom: 150px;
  }
  .navbar {
    margin-top: 155px;
  }
}

.delimiter {
  position: absolute;
  top: 0;
  left: 50%;

  width: 1px;
  height: 320px;
  background-color: $mercuryLight;

  transform: translateX(-50%);
}

