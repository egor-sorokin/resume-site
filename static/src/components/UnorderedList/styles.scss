@import '../../styles/base/_vars.scss';

.unorderedlist {
  &__item {
    text-align: center;

    &:nth-of-type(n+2) {
      margin-top: 8px;
    }
  }
  &__text {
    color: $mercuryLight;
    font-size: 16px;
    font-weight: 300;
  }
}
