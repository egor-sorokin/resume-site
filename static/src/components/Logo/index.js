import React from 'react';
import './styles.scss';


const Logo = () => (
  <div className="logo">
    <svg className="logo__circles" height="87" viewBox="0 0 88 87" width="88">
      <filter id="a" height="107.2%" width="104.7%" x="-2.3%" y="-2.4%">
        <feOffset dy="2" in="SourceAlpha" result="shadowOffsetOuter1"/>
        <feColorMatrix in="shadowOffsetOuter1" result="shadowMatrixOuter1"
                       values="0 0 0 0 0   0 0 0 0 0   0 0 0 0 0  0 0 0 0.5 0"/>
        <feMerge>
          <feMergeNode in="shadowMatrixOuter1"/>
          <feMergeNode in="SourceGraphic"/>
        </feMerge>
      </filter>
      <g fill="none" fillRule="evenodd" filter="url(#a)" stroke="#ddd" transform="translate(1 1)">
        <ellipse cx="43" cy="41.5" fill="#424242" rx="43" ry="41.5"/>
        <ellipse cx="53.238" cy="27.553" fill="#2d2b28" rx="25.254" ry="24.832"/>
        <ellipse cx="61.429" cy="18.709" fill="#696969" rx="13.651" ry="13.266"/>
        <ellipse cx="57.675" cy="65.992" fill="#696969" rx="13.992" ry="13.607"/>
        <ellipse cx="61.429" cy="70.074" fill="#2d2b28" rx="8.19" ry="8.164"/>
        <ellipse cx="15.016" cy="39.799" fill="#696969" rx="15.016" ry="14.627"/>
        <ellipse cx="26.619" cy="39.799" rx="26.619" ry="25.512"/>
        <ellipse cx="8.19" cy="40.139" fill="#909090" rx="8.19" ry="8.164"/>
      </g>
    </svg>
  </div>
);

export default Logo;
