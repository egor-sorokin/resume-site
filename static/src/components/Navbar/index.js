import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const Navbar = ({items}) => {
  const navbarItems = items.map((item) => (
    <li className="navbar__item">
      <a href="/" className="navbar__text">{item}</a>
    </li>
  ));

  return (
    <ul className="navbar">
      {navbarItems}
    </ul>
  )
};

Navbar.propTypes = {
  items: PropTypes.instanceOf(Array)
};

Navbar.defaultProps = {
  items: []
};

export default Navbar;