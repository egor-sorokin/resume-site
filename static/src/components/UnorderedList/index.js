import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const UnorderedList = ({title, items}) => {
  const unorderedListItems = items.map((item) => (
    <li key={item.id}
        className="unorderedlist__item">
      <a href={item.url}
         target="_blank"
         className="unorderedlist__text">{item.name}</a>
    </li>
  ));

  return (
    <div className="unorderedlist">
      {title.length > 0 && (<h5 className="unorderedlist__title">{title}</h5>)}
      <ul className="unorderedlist__list">
        {unorderedListItems}
      </ul>
    </div>
  )
};

UnorderedList.propTypes = {
  title: PropTypes.string,
  items: PropTypes.instanceOf(Array)
};

UnorderedList.defaultProps = {
  title: '',
  items: []
};

export default UnorderedList;
