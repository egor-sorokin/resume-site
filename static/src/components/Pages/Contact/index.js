import React from 'react';
import PropTypes from 'prop-types';
import Logo from '../../Logo/index';
import UnorderedList from '../../UnorderedList/index';
import Navbar from '../../Navbar/index';
import withFetching from '../../../utils/api';
import {URL_PATH_PERSON_DATA} from '../../../constants/index';
import './styles.scss';


const Contact = ({data, isFetching, error}) => {
  const personData = data.personData || {};
  const socials = personData.socials || [];
  const navbarItems = ['Home', 'About', 'Projects', 'Experiments'];

  if (error) {
    return <div><p>{error}</p></div>;
  }

  if (isFetching) {
    return <div><p>Loading...</p></div>;
  }

  return (
    <div>
      <section className="banner contact">
        <div className="contact-inner">
          <div className="contact-inner__item">
            <Navbar
              items={navbarItems}
            ></Navbar>
            <div className="delimiter"></div>
          </div>
          <div className="contact-inner__item">
            <Logo></Logo>
          </div>
          <div className="contact-inner__item">
            <UnorderedList
              title=''
              items={socials.items}
            ></UnorderedList>
          </div>
        </div>
      </section>
    </div>
  );
};

Contact.propTypes = {
  data: PropTypes.shape({}),
  isFetching: PropTypes.bool,
  error: PropTypes.string,
};

Contact.defaultProps = {
  data: {},
  error: '',
  isFetching: false,
};


export default withFetching(URL_PATH_PERSON_DATA, Contact);
