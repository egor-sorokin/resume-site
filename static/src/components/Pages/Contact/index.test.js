// eslint-disable-next-line
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import Contact from './index';
import Logo from '../../Logo/index';
import UnorderedList from '../../UnorderedList/index';
import Navbar from '../../Navbar/index';


describe('<Contact />', () => {
  let component;
  const personData = {
    id: 1,
    first_name: 'Bob',
    last_name: 'Cat',
    email: 'example@admin.com',
    summary: [
      {
        title: 'test title',
        text: 'test text'
      },
      {
        title: 'test title 2',
        text: 'test text 2'
      }
    ],
    socials: {
      title: 'test title',
      items: [
        {
          id: 1,
          text: 'test text 1',
          url: 'test.urlone'
        },
        {
          id: 2,
          text: 'test text 2',
          url: 'test.urltwo'
        }
      ]
    }
  };


  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<Contact />);
    });


    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });


    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 1 components', () => {
      expect(component.children()).toHaveLength(1);
    });
  });


  describe('props tests', () => {
    it('renders data', () => {
      component = mount(<Contact />);
      component.setProps({personData: personData});
      expect('personData' in component.props()).toBeTruthy();
      expect(component.find(<Navbar />)).toBeTruthy();
      expect(component.find(<Logo />)).toBeTruthy();
      expect(component.find(<UnorderedList />)).toBeTruthy();

      component.setProps({personData: {}});
    });
  });
});
