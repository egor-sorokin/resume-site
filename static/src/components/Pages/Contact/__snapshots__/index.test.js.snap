// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> base tests matches snapshot 1`] = `
<div>
  <section
    className="banner contact"
  >
    <div
      className="contact-inner"
    >
      <div
        className="contact-inner__item"
      >
        <Navbar
          items={
            Array [
              Object {
                "text": "Home",
                "type": "white",
                "url": "#home",
              },
              Object {
                "text": "Projects",
                "type": "white",
                "url": "#projects",
              },
              Object {
                "text": "Experiments",
                "type": "white",
                "url": "#experiments",
              },
            ]
          }
        />
        <div
          className="delimiter"
        />
      </div>
      <div
        className="contact-inner__item"
      >
        <Logo />
      </div>
      <div
        className="contact-inner__item"
      >
        <UnorderedList
          items={Array []}
          title=""
          type="mask"
        />
      </div>
    </div>
  </section>
</div>
`;
